Fabric.Authorization.API/Modules/GroupsModule.cs

```diff
@@ -165,32 +178,36 @@ public class GroupsModule : FabricModule<Group>
             }
         }
 
-        private async Task<dynamic> AddRoleToGroup()
+        private async Task<dynamic> AddRolesToGroup(dynamic parameters)
         {
-            try
+            var apiRoles = this.Bind<List<RoleApiModel>>();
+            var errorResponse = await ValidateRoles(apiRoles);
+            if (errorResponse != null)
             {
-                this.RequiresClaims(AuthorizationWriteClaim);
-                var groupRoleRequest = this.Bind<GroupRoleRequest>();
-                if (groupRoleRequest.Id == null && groupRoleRequest.RoleId == null)
-                {
-                    return CreateFailureResponse("Role ID is required.", HttpStatusCode.BadRequest);
-                }
+                return errorResponse;
+            }
 
-                var roleId = groupRoleRequest.Id ?? groupRoleRequest.RoleId;
-                var group = await _groupService.AddRoleToGroup(groupRoleRequest.GroupName, roleId.Value);
-                return CreateSuccessfulPostResponse(group.ToGroupRoleApiModel());

```

Comment:
Not needed for this PR, but maybe `CheckWriteAccess` can be changed in the future to accept a collection of tuples containing grain/securableItem pairs and it can loop over them internally.

Reply:
not a bad idea